cmd/rootMain.go

```diff
@@ -59,6 +59,26 @@ func rootMain(cmd *cobra.Command, args []string) exitCode {
 	// Set plugin interactivity to false if execfile or rootCommandFlag were specified.
 	plugin.InitInteractive(execfile == "" && rootCommandFlag == "")
 
+	// If interactive and this process is in its own process group, then fork and run the original
+	// command as a new process that's not the leader of its process group. This is specifically to
+	// work with shells (zsh) that try to use their parent's process group, which would no longer be
+	// in the same session. By forking, we keep the original process group in its original session so
+	// the child shell can still modify it.
+	if plugin.IsInteractive() && os.Getpid() == unix.Getpgrp() {
```

Comment:
From what I'm seeing in https://linux.die.net/man/2/setsid, could the reasoning behind this be simplified to "we ensure that the original command is not the leader of its process group so that `setsid` (which creates the new session) works"?

Reply:
Kind of, bet technically the other approach would have also worked (and we wouldn't need to exec).